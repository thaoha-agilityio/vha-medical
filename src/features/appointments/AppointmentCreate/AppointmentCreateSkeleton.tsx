import { Button, InputSearch } from '@/components/ui';

export const AppointmentCreateSkeleton = () => (
  <div className="flex mt-3 justify-between gap-10 mb-8">
    <InputSearch placeholder="Search Appointments" isDisabled={true} />
    <Button className="h-[52px] font-medium" isDisabled={true}>
      Create
    </Button>
  </div>
);
