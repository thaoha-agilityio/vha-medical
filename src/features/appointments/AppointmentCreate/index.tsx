'use client';

import { memo } from 'react';
import { useDisclosure } from '@nextui-org/react';

// Components
import { Button } from '@/components/ui';
import AppointmentModal from '../AppointmentModal';

// Types
import { UserLogged } from '@/types';

export interface AppointmentCreateProps {
  userLogged: UserLogged | null;
}
const AppointmentCreate = ({ userLogged }: AppointmentCreateProps) => {
  const { onClose, onOpen, isOpen } = useDisclosure();

  return (
    <>
      <Button onClick={onOpen} className="h-[52px] font-medium">
        Create
      </Button>
      {isOpen && (
        <AppointmentModal
          userLogged={userLogged}
          onClose={onClose}
          isOpen={isOpen}
        />
      )}
    </>
  );
};

export default memo(AppointmentCreate);
