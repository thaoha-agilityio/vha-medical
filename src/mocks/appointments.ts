import { AVATAR_THUMBNAIL } from '@/constants';
import { APIResponse, AppointmentModel } from '@/types';

export const MOCK_APPOINTMENTS: APIResponse<AppointmentModel>[] = [
  {
    id: '1',
    attributes: {
      startTime: '2026-09-11T06:30:00.000Z',
      durationTime: '01:30:00',
      status: 0,
      receiverId: {
        data: {
          id: '2',
          attributes: {
            username: 'Jessica Jane',
            email: 'jessicajane@gmail.com.vn',
            description:
              "Hi, I'm Jessica Jane. Lorem Ipsum is simply dummy text of the printing and typesetting for it.",
            rating: 5,
            tasks: 40,
            reviews: 600,
            avatar: AVATAR_THUMBNAIL,
          },
        },
      },
      senderId: {
        data: {
          id: '3',
          attributes: {
            username: 'Alex Stanton',
            email: 'alexstanton@gmail.com',
            description:
              "Hi, I'm Alex Stanton. Happy for all problem happen with me. Let's smile.",
            rating: 5,
            tasks: 40,
            reviews: 600,
            avatar: AVATAR_THUMBNAIL,
          },
        },
      },
    },
  },
  {
    id: '2',
    attributes: {
      startTime: '2026-09-12T02:00:00.000Z',
      durationTime: '01:00:00',
      status: 0,
      receiverId: {
        data: {
          id: '6',
          attributes: {
            username: 'ThangHq',
            email: 'thang.hoquang@asnet.com.vn',
            description:
              "Hi, I'm Thang Ho Quang. I'm a software engineer in AgilityIO company. I'm so happy for my job.",
            rating: 5,
            tasks: 40,
            reviews: 600,
            avatar: AVATAR_THUMBNAIL,
          },
        },
      },
      senderId: {
        data: {
          id: '3',
          attributes: {
            username: 'Alex Stanton',
            email: 'alexstanton@gmail.com',
            description:
              "Hi, I'm Alex Stanton. Happy for all problem happen with me. Let's smile.",
            rating: 5,
            tasks: 40,
            reviews: 600,
            avatar: AVATAR_THUMBNAIL,
          },
        },
      },
    },
  },
  {
    id: '3',
    attributes: {
      startTime: '2024-09-13T03:00:00.000Z',
      durationTime: '01:00:00',
      status: 1,
      receiverId: {
        data: {
          id: '3',
          attributes: {
            username: 'Alex Stanton',
            email: 'alexstanton@gmail.com',
            description:
              "Hi, I'm Alex Stanton. Happy for all problem happen with me. Let's smile.",
            rating: 5,
            tasks: 40,
            reviews: 600,
            avatar: AVATAR_THUMBNAIL,
          },
        },
      },
      senderId: {
        data: {
          id: '6',
          attributes: {
            username: 'ThangHq',
            email: 'thang.hoquang@asnet.com.vn',
            description:
              "Hi, I'm Thang Ho Quang. I'm a software engineer in AgilityIO company. I'm so happy for my job.",
            rating: 5,
            tasks: 40,
            reviews: 600,
            avatar: AVATAR_THUMBNAIL,
          },
        },
      },
    },
  },
];
