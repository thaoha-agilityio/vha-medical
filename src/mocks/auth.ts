export const MOCK_AUTH = {
  EMAIL: 'user@example.com',
  PASSWORD: 'Password123',
};
