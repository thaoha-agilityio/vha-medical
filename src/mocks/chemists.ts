import { AVATAR_THUMBNAIL } from '@/constants';
import { APIResponse, ChemistModel, SpecialtyResponse } from '@/types';

export const MOCK_CHEMISTS_LIST: Array<APIResponse<ChemistModel>> = [
  {
    id: '6',
    attributes: {
      users_permissions_user: {
        data: {
          id: '8',
          attributes: {
            username: 'Alex Stanton',
            email: 'alexstanton@gmail.com',
            description:
              "Hi, I'm Alex Stanton. Happy for all problem happen with me. Let's smile.",
            rating: 5,
            tasks: 22,
            reviews: 321,
            avatar: AVATAR_THUMBNAIL,
            specialtyId: {
              data: {
                id: '8',
                attributes: {
                  name: 'Laboratory Chemist',
                },
              },
            },
          },
        },
      },
    },
  },
  {
    id: '7',
    attributes: {
      users_permissions_user: {
        data: {
          id: '9',
          attributes: {
            username: 'Anna White',
            email: 'anna_white@asnet.com.vn',
            description: '',
            rating: 5,
            tasks: 40,
            reviews: 600,
            avatar: AVATAR_THUMBNAIL,
            specialtyId: {
              data: {
                id: '10',
                attributes: {
                  name: 'Power Plant Chemist',
                },
              },
            },
          },
        },
      },
    },
  },
  {
    id: '8',
    attributes: {
      users_permissions_user: {
        data: {
          id: '7',
          attributes: {
            username: 'Jessica Jane',
            email: 'jessicajane@gmail.com.vn',
            description:
              "Hi, I'm Jessica Jane. Lorem Ipsum is simply dummy text of the printing and typesetting for it.",
            rating: 4,
            tasks: 23,
            reviews: 123,
            avatar: AVATAR_THUMBNAIL,
            specialtyId: {
              data: {
                id: '9',
                attributes: {
                  name: 'Instrumentation',
                },
              },
            },
          },
        },
      },
    },
  },
  {
    id: '9',
    attributes: {
      users_permissions_user: {
        data: {
          id: '10',
          attributes: {
            username: 'John Smith',
            email: 'john_smith@asnet.com.vn',
            description: "I'm John Smith from VHA with love.",
            rating: 3,
            tasks: 12,
            reviews: 333,
            avatar: AVATAR_THUMBNAIL,
            specialtyId: {
              data: {
                id: '7',
                attributes: {
                  name: 'QC Chemist',
                },
              },
            },
          },
        },
      },
    },
  },
  {
    id: '10',
    attributes: {
      users_permissions_user: {
        data: {
          id: '6',
          attributes: {
            username: 'ThangHq',
            email: 'thang.hoquang@asnet.com.vn',
            description:
              "Hi, I'm Thang Ho Quang. I'm a software developer in AgilityIO. I'm so happy my job.",
            rating: 5,
            tasks: 40,
            reviews: 600,
            avatar: AVATAR_THUMBNAIL,
            specialtyId: {
              data: {
                id: '6',
                attributes: {
                  name: 'Organic Chemist',
                },
              },
            },
          },
        },
      },
    },
  },
];

export const MOCK_SPECIALTIES: SpecialtyResponse[] = [
  {
    id: '1',
    attributes: {
      name: 'Instrumentation',
      type: 1,
    },
  },
  {
    id: '2',
    attributes: {
      name: 'Laboratory Chemist',
      type: 2,
    },
  },
  {
    id: '3',
    attributes: {
      name: 'Organic Chemist',
      type: 3,
    },
  },
  {
    id: '4',
    attributes: {
      name: 'Power Plant Chemist',
      type: 4,
    },
  },
  {
    id: '5',
    attributes: {
      name: 'QC Chemist',
      type: 5,
    },
  },
];
