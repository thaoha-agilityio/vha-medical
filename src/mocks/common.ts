export const MOCK_PAGINATION = {
  page: 1,
  pageCount: 3,
  pageSize: 6,
  total: 13,
};
