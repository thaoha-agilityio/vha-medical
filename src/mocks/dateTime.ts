export const MOCK_DATE = {
  INVALID: 'Invalid Date',
  NOW: '2024-02-11T10:00:00.000',
  PASS: '2022-09-20T02:36:00.000',
  FUTURE: '2024-09-18T02:30:00.000',
};
