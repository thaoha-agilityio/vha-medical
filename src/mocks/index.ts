export * from './appointments';
export * from './auth';
export * from './chemists';
export * from './common';
export * from './dateTime';
export * from './notification';
export * from './options';
export * from './profile';
export * from './table';
export * from './user';
