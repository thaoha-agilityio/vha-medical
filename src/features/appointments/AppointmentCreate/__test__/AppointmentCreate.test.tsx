import { act, render, renderHook, screen } from '@testing-library/react';
import { useDisclosure } from '@nextui-org/react';

import AppointmentCreate, { AppointmentCreateProps } from '..';
import { MOCK_USERS_LOGGED } from '@/mocks';

describe('AppointmentCreate', () => {
  const mockProps: AppointmentCreateProps = {
    userLogged: MOCK_USERS_LOGGED[0],
  };

  const setup = async (props: AppointmentCreateProps) =>
    act(() => render(<AppointmentCreate {...props} />));

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', async () => {
    const { asFragment } = await setup(mockProps);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders AppointmentModal with correct props when open', async () => {
    const { result } = renderHook(() => useDisclosure());
    await setup(mockProps);

    const createButton = screen.getByRole('button', {
      name: /create/i,
    });

    createButton.click();

    expect(result.current.onOpen).toBeTruthy();
  });
});
