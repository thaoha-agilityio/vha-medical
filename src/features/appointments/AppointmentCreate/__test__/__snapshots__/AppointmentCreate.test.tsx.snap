// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppointmentCreate should render correctly 1`] = `
<DocumentFragment>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover border-transparent bg-linear-success text-content1 h-[52px] font-medium"
    type="button"
  >
    Create
  </button>
</DocumentFragment>
`;
